packages/editor/src/components/provider/index.js

```diff
@@ -161,8 +165,17 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 		__unstableTemplate: template,
 	} ) => {
 		const mode = useSelect(
-			( select ) => select( editorStore ).getRenderingMode(),
-			[]
+			( select ) => {
+				const postTypeObject = select( coreStore ).getPostType(
+					post.type
+				);
+				return (
+					postTypeObject?.default_rendering_mode ??
+					settings.defaultRenderingMode ??
```

Comment:
Two questions here: 

1- Should we remove the settings.defaultRenderMode from here ? 
2- Should we avoid rendering the editor until the postType object is loaded (to avoid having a jump in rendering mode)

Reply:
@youknowriad 

> Should we remove the settings.defaultRenderMode from here ?

I don't think it hurts to keep `settings.defaultRenderingMode` really for backwards compatibility or giving another place to define the default, but I can remove it if we feel it's unnecessary. Perhaps it's better to have a single source of truth rather than having multiple ways to define it?

> Should we avoid rendering the editor until the postType object is loaded (to avoid having a jump in rendering mode)

I think that's probably a good idea. I've added a check that `mode` isn't `falsey` with the `isReady` check, so it will return `null` until `mode` is available.

Reply:
> Perhaps it's better to have a single source of truth rather than having multiple ways to define it?

I think this was not really an API third-parties were able to use easily, so I'd be for removing it and keeping things simple (if that's possible :P).